### ISSUE TITLE
Lacking Clarity (and Consistency): Table of Symbols Constructors

### ISSUE THREAD

**Issue by Mornix** [JacquesCarette/Drasil#1658]
Table of Symbols Constructors are in an odd place.

https://github.com/JacquesCarette/Drasil/blob/f43b3c189ac90af4681e5f1250077817a0f2e86a/code/drasil-docLang/Drasil/DocumentLanguage.hs#L78-L83

https://github.com/JacquesCarette/Drasil/blob/f43b3c189ac90af4681e5f1250077817a0f2e86a/code/drasil-docLang/Drasil/DocumentLanguage.hs#L119-L124

The Constructors seem fairly logical. A simple constructor, and a more customizable one. While this is something that could benefit from a smart constructor, it isn't something of super high priority.

If we look at what the constructors do to create the table of symbols, the lack of clarity (should) become[s] clear. 

https://github.com/JacquesCarette/Drasil/blob/f43b3c189ac90af4681e5f1250077817a0f2e86a/code/drasil-docLang/Drasil/DocumentLanguage.hs#L267-L274

https://github.com/JacquesCarette/Drasil/blob/f43b3c189ac90af4681e5f1250077817a0f2e86a/code/drasil-docLang/Drasil/DocumentLanguage.hs#L279-L293

https://github.com/JacquesCarette/Drasil/blob/f43b3c189ac90af4681e5f1250077817a0f2e86a/code/drasil-database/Database/Drasil/SystemInformation.hs#L23-L31

A `TSymb` constructor takes all members of the `_quant` field to print. This can be something such as a `[QuantityDict]` or `[DefinedQuantityDict]`. Nothing too wrong with this constructor, although it could be a great candidate for automation.

Looking that the other constructor, `TSymb'` (and related function -- `mkTSymb`), we can see that an additional constraint is required: `Concept`. As a result of this, the `_quants` field is not pulled as it doesn't contain the `Concept` constraint, instead the `_concepts` field of `SystemInformation` is used. With this being the case, we cannot enumerate symbols which are missing a `Concept`. This alienates `QuantityDict`, `UnitaryChunk` (which GlassBR makes heavy use of), among others. The upside is, with the `TSymb'` constructor, symbols are pulled automatically from the document.

This restriction seems a (little) arbitrary and hacky, especially if `LFunc` is `Term`. We are left with some symbols (possibly) being excluded. Moreover, it seems odd that if we want to describe the `term` of a symbol we can get "all" symbols, whereas when we describe `defn` we may only display a subset. 

TL;DR: Is it intentional that `TSymb'` takes a subset of all symbols and thus a Table of Symbols would be "incomplete"?

**Comment by JacquesCarette** [comment]
I would be very surprised if this was intentional. [Well spotted BTW].  This likely evolved organically, and some of the divergence happened "to make things work".

This Quantity vs Quantity+Concept thing has happened multiple times already. I wonder how many Quantities that aren't Concept we actually have.

**Comment by samm82** [comment]
Just a note that the `quantPlate` branch still needs some work and should be considered while working on this issue.
